### PR TITLE
Prompt Tuning

### PR DESCRIPTION
**Pull request by danielbank**
- Moving knowledge base to a separate directory, `/corpus`
- Loading all table schemas into each test generation to prevent schema errors
- Setting temperature to 0.0 for consistent test generation
- Adding good / bad examples to steer the model away from generating broken tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,9 +72,21 @@ supabase link
 
 - This command runs all SQL test files located in the `supabase/tests` directory.
 
-## Contributing
+## Model Providers
+
+Currently, the only supported model is Claude because Antropic provided free credits for the Supabase AI Hackathon in which this project was developed. If you prefer to use a different model provider like ChatGPT, LLaMa, or Gemini, please submit a pull request.
```
Comment (ariel-nathan) [inline README.md]: Typo on `Antropic` -> Anthropic

File: src/worker.ts [hunk 2]
```diff
@@ -42,22 +39,19 @@ async function getTableSchema(
       INNER JOIN information_schema.columns c ON
         t.table_name = c.table_name
       WHERE
-        t.table_schema = $1
+        t.table_schema = 'public'
```
Comment (ariel-nathan) [inline src/worker.ts]: What if the table schema is not `public`. For example `storage.objects`

File: supabase/migrations/20241123163421_posts-and-comments.sql [hunk 3]
```diff
@@ -24,15 +24,17 @@ create policy "Users can CRUD their own posts"
...
+    using (exists (select 1 from posts where id = post_id and is_published));
 
+-- This policy is flawed because the user is not constrained from posting as another user
+-- To fix the security vulnerability, we need to add: `and created_by=auth.uid()` to the `with check` clause
 create policy "Users can write comments on posts they can view"
-    on comments as permissive for insert to authenticated
-    with check (exists (select 1 from posts where id = post_id and is_published is true));
\ No newline at end of file
+    on comments for insert to authenticated
+    with check (exists (select 1 from posts where id = post_id and is_published) and created_by=auth.uid());
```
Comment (ariel-nathan) [inline supabase/migrations/20241123163421_posts-and-comments.sql]: Did you mean to include `and created_by=auth.uid()`?

**Review by ariel-nathan — APPROVED** [review]
Looks good!

**Review by Kelach — APPROVED** [review]
Looks good.

**Pull request merged**